frontend/src/scenes/notebooks/Nodes/NotebookNodeSurvey.tsx

```diff
@@ -13,14 +13,18 @@ import { StatusTag } from 'scenes/surveys/Surveys'
 import { SurveyResult } from 'scenes/surveys/SurveyView'
 import { SurveyAppearance } from 'scenes/surveys/SurveyAppearance'
 import { SurveyReleaseSummary } from 'scenes/surveys/Survey'
-import api from 'lib/api'
+import { useEffect } from 'react'
 
 const Component = (props: NotebookNodeViewProps<NotebookNodeSurveyAttributes>): JSX.Element => {
     const { id } = props.attributes
     const { survey, surveyLoading, hasTargetingFlag } = useValues(surveyLogic({ id }))
     const { expanded, nextNode } = useValues(notebookNodeLogic)
     const { insertAfter } = useActions(notebookNodeLogic)
 
+    useEffect(() => {
+        props.updateAttributes({ title: survey.name ? `Survey: ${survey.name}` : 'Survey' })
```

Comment:
Do we know if under the hood TipTap diffs to determine if the property has actually changed? My worry would be that this will recompute every time the component is mounted (so every notebook load) which could trigger the notebook to re-save if the content is updated through `onUpdate`

Reply:
I had the same thought and tested it out. It indeed does diff so setting the same string is basically a no-op 👍